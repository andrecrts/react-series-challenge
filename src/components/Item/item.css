.contentItem {
  display: inline-block;
  position:relative;
  width: 200px;
  height: 200px;
  margin: 1em;
}

.imgContent {
  width:100%;
  height:100%;
}

.favButton {
  position: absolute;
  top:0;
  right:0;
  padding: 2px 2px 0 0;
  width: 24px;
  height: 24px;
}
