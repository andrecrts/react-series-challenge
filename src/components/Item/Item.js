import React from 'react'
import favIcon from '../../assets/fav.svg'
import noFavIcon from '../../assets/no-fav.svg'
import './item.css'

const Item = ({item, onClickFav, isFav}) => {

  const favImage = isFav ? favIcon : noFavIcon
  return(
    <div className="item">
      <img src={favImage} className="favButton" onClick={() => onClickFav(item, isFav)} alt="Favorite"/>
      <img src={item.images.original.webp} className="imgContent" alt={item.title}/>
    </div>
  )
}

export default Item
