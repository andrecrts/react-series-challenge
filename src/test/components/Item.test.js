import React from 'react'
import { shallow } from 'enzyme'
import Item from '../../components/Item'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'


const item= {
  title:'title1', images:{original:{webp:'srcImage'}}
}
test('Home snapshot', () => {
  const onClick = (item, isFav) => {}
  const component = shallow(
    <Item
      item={item}
      onClick={onClick}
      onClickFav={onClick}
    />
  )
  expect(toJson(component)).toMatchSnapshot();
})


// describe('test', () => {
//   it('will work', () => {
//     const onClick = (item, isFav) => {}
//     const component = shallow(
//       <Item
//         item={item}
//         onClick={onClick}
//         onClickFav={onClick}
//       />
//     )
//     expect(component.text()).toBe('hola')
//   })
// })

