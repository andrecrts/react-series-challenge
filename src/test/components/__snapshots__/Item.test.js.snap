// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div
  className="sc-bdVaJa jFnFkT"
>
  <img
    alt="Favorite"
    className="sc-bwzfXH caOhOO"
    onClick={[Function]}
    src="no-fav.svg"
  />
  <img
    alt="title1"
    className="sc-htpNat kewJQA"
    onClick={[Function]}
    src="srcImage"
  />
</div>
`;
